app/models/foreman_tasks/task.rb

```diff
@@ -225,6 +225,28 @@ def to_label
       parts.join(' ').strip
     end
 
+    def self.get_latest_tasks_by_resource_ids(label, resource_ids, resource_type)
+      ids = resource_ids.join(',')
+      base_combined_table = search_for("label = #{label} and resource_type = #{resource_type} and resource_id ^ (#{ids})")
+
+      grouped = base_combined_table.select(
+        "MAX(#{self::DynflowTask.table_name}.started_at) AS started_at",
+        "#{ForemanTasks::Lock.table_name}.resource_id AS resource_id"
+      ).group("#{ForemanTasks::Lock.table_name}.resource_id")
+
+      max_per_resource_id = self::DynflowTask.joins(
+        :locks
+      ).select(
+        "#{self::DynflowTask.table_name}.*, inner_select.resource_id"
+      ).joins(
+        "INNER JOIN (#{grouped.to_sql}) inner_select
```

Comment:
Shouldn't it be `INNER JOIN ($subquery) AS inner_select`? Or is the `AS` actually optional?

Reply:
As far as I know with tables, which is how the subquery is used here, the `AS` is not necessary.
See also https://www.postgresql.org/docs/10/sql-select.html:
```
where from_item can be one of:

    [ ONLY ] table_name [ * ] [ [ AS ] alias [ ( column_alias [, ...] ) ] ]
...
    from_item [ NATURAL ] join_type from_item [ ON join_condition | USING ( join_column [, ...] ) ]

Reply:
Good to know, thanks